src/clp_notification_monitor/main.py

```diff
@@ -70,11 +77,59 @@ def submit_compression_jobs_thread_entry(
         while True:
             compression_buffer.wait_for_compression_jobs()
             sleep_time: int = 1
-            while False is compression_buffer.try_compress_from_fs():
-                time.sleep(sleep_time)
-                sleep_time = min(sleep_time * 2, max_polling_period)
+            while True:
+                paths_to_compress = compression_buffer.get_paths_to_compress()
+                if len(paths_to_compress) > 0:
+                    new_job_entry: Dict[str, Any]
+                    if "s3" == input_type:
+                        input_buckets = []
+                        for full_s3_path in paths_to_compress:
+                            input_buckets.append(
+                                {
+                                    "endpoint_url": seaweed_s3_endpoint_url,
+                                    "s3_path_prefix": str(full_s3_path),
+                                    # Remove the bucket name from the path prefixes
+                                    "s3_path_prefix_to_remove_from_mount": "".join(
+                                        full_s3_path.parts[:2]
+                                    ),
+                                }
+                            )
+                        new_job_entry = {
+                            "input_type": "s3",
+                            "input_config": {
+                                "access_key_id": "",  # not used
+                                "secret_access_key": "",  # not used
+                                "buckets": input_buckets,
+                            },
+                            "output_config": {},
+                            "status": "pending",
+                            "submission_timestamp": math.floor(time.time() * 1000),
+                        }
+                    else:
+                        fs_compression_paths = []
+                        for full_s3_path in paths_to_compress:
+                            mounted_path_str = seaweed_mnt_prefix + str(full_s3_path)
+                            fs_compression_paths.append(mounted_path_str)
+                        new_job_entry = {
+                            "input_type": "fs",
+                            "input_config": {
+                                "path_prefix_to_remove": seaweed_mnt_prefix,
+                                "paths": fs_compression_paths,
+                            },
+                            "output_config": {},
+                            "status": "pending",
+                            "submission_timestamp": math.floor(time.time() * 1000),
+                        }
+
+                    jobs_collection.insert_one(new_job_entry)
+                    logger.info("Submitted job to compression database.")
+                    break
+                else:
+                    time.sleep(sleep_time)
+                    sleep_time = min(sleep_time * 2, max_polling_period)
```

Comment:
```suggestion
            while True:
                paths_to_compress: List[PurePath] = compression_buffer.get_paths_to_compress()
                if 0 == len(paths_to_compress):
                    time.sleep(sleep_time)
                    sleep_time = min(sleep_time * 2, max_polling_period)
                    continue
                new_job_entry: Dict[str, Any]
                if "s3" == input_type:
                    input_buckets: List[Dict[str, Any]] = []
                    for full_s3_path in paths_to_compress:
                        input_buckets.append(
                            {
                                "endpoint_url": seaweed_s3_endpoint_url,
                                "s3_path_prefix": str(full_s3_path),
                                # Remove the bucket name from the path prefixes
                                "s3_path_prefix_to_remove_from_mount": "".join(
                                    full_s3_path.parts[:2]
                                ),
                            }
                        )
                    new_job_entry = {
                        "input_type": "s3",
                        "input_config": {
                            "access_key_id": "",  # not used
                            "secret_access_key": "",  # not used
                            "buckets": input_buckets,
                        },
                        "output_config": {},
                        "status": "pending",
                        "submission_timestamp": math.floor(time.time() * 1000),
                    }
                else:
                    fs_compression_paths: List[str] = []
                    for full_s3_path in paths_to_compress:
                        mounted_path_str: str = seaweed_mnt_prefix + str(full_s3_path)
                        fs_compression_paths.append(mounted_path_str)
                    new_job_entry = {
                        "input_type": "fs",
                        "input_config": {
                            "path_prefix_to_remove": seaweed_mnt_prefix,
                            "paths": fs_compression_paths,
                        },
                        "output_config": {},
                        "status": "pending",
                        "submission_timestamp": math.floor(time.time() * 1000),
                    }

                jobs_collection.insert_one(new_job_entry)
                logger.info("Submitted job to compression database.")
                break
```

Reply:
1. Type annotation
2. Try to avoid nested if-else
Please run the linter just in case I make any mistakes